module/zfs/vdev_queue.c

```diff
@@ -936,14 +936,13 @@ vdev_queue_io_done(zio_t *zio)
 	vdev_queue_t *vq = &zio->io_vd->vdev_queue;
 	zio_t *nio;
 
-	mutex_enter(&vq->vq_lock);
+	hrtime_t now = gethrtime();
+	vq->vq_io_complete_ts = now;
+	vq->vq_io_delta_ts = zio->io_delta = now - zio->io_timestamp;
```

Comment:
I think that these assignments into `vq` should still take place under the `vq_lock`

Reply:
As I see, these variables are read without holding the vq_lock, so if they wish to get torn on 32-bit system, they will, and vq_lock won't help.